marklogic-client-api/src/main/java/com/marklogic/client/impl/NodeConverter.java

```diff
@@ -409,6 +409,9 @@ static public ReaderHandle ReaderToHandle(Reader value) {
    static public Stream<ReaderHandle> ReaderToHandle(Stream<? extends Reader> values) {
       return (values == null) ? null : values.map(NodeConverter::ReaderToHandle);
    }
+   static public ReaderHandle[] ReaderToHandle(Reader[] values) {
+       return (values == null) ? null : (Stream.of(values).map(value-> new ReaderHandle(value)).toArray(ReaderHandle[]::new));
+    }
```

Comment:
The Reader[] to ReaderHandle[] conversion is just one of the data types that must be converted to handles.

I've added an addendum to https://project.marklogic.com/jira/browse/JAVA-62 to propose an approach.

Please take a look and see what you think.